### PR TITLE
snowflake-snowpark-python 1.26.0

### PR DESCRIPTION
**Pull request by lorepirri**
snowflake-snowpark-python 1.26.0

**Destination channel:** defaults

### Links

- [PKG-6475]
- dev_url (pypi): https://github.com/snowflakedb/snowpark-python/tree/v1.26.0
- diff: https://github.com/snowflakedb/snowpark-python/compare/v1.25.0...v1.26.0
- conda-forge: https://github.com/conda-forge/snowflake-snowpark-python-feedstock/blob/main/recipe/meta.yaml
- pypi: https://pypi.org/project/snowflake-snowpark-python/1.26.0
- pypi inspector: https://inspector.pypi.io/project/snowflake-snowpark-python/1.26.0

### Explanation of changes:

- bump version

[PKG-6475]: https://anaconda.atlassian.net/browse/PKG-6475?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -24,18 +24,19 @@ requirements:
     - pip
     - setuptools >=40.6.0
     - wheel
-    # Required to compile .proto files during setup.py execution
-    # protoc is provided by protobuf. We do not have a package protoc-wheel.
-    - protobuf
+    # Snowpark IR
```
Comment (msentissi) [inline recipe/meta.yaml:27]: What is the rationale behind this change ?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -24,18 +24,19 @@ requirements:
...
+    - protobuf ==3.20.1  # [py<=310]
+    - protobuf ==4.25.3  # [py>310]
```
Comment (msentissi) [inline recipe/meta.yaml:29]: What is the rationale behind this change ?

Comment (lorepirri) [reply recipe/meta.yaml:29]: They did update their recipe file with those pinnings. Even though that file is not used by us or them, it probably represents their intention to pin `protobuf` to those versions for some reason. They probably could have used `setup_requires` for that.

https://github.com/snowflakedb/snowpark-python/blob/v1.26.0/recipe/meta.yaml#L29-L32

Comment (lorepirri) [reply recipe/meta.yaml:32]: They ditched `protoc` and therefore I also removed the comment above `protobuf`.

https://github.com/snowflakedb/snowpark-python/blob/v1.26.0/setup.py#L28

Comment (lorepirri) [reply recipe/meta.yaml:32]: I added a reference comment.

Comment (msentissi) [reply recipe/meta.yaml:32]: I see, that makes sense. Thank you !

**Review by msentissi — APPROVED** [review]

**Review by SC426 — APPROVED** [review]
